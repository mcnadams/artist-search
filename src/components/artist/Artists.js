import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

function Artists({ artists }) {
  //get artists from search container
  const artistList = artists.map((artist, i) => {
    return (
      <li key={i}>
        <Artist artist={artist} />
      </li>
    );
  });

  return (
    <ul style={{ background: 'white', listStyle: 'none', width: '15vw' }}>
      {artistList}
    </ul>
  );
}

Artists.propTypes = {
  artists: PropTypes.array.isRequired
};

export default Artists;
