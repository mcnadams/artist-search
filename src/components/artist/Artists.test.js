import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists renders', () => {
  it('renders Artists', () => {
    const wrapper = shallow(<Artists artists={[{ name: 'lady' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
