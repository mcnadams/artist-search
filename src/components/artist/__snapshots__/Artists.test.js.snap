// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists renders renders Artists 1`] = `
<ul
  style={
    Object {
      "background": "white",
      "listStyle": "none",
      "width": "15vw",
    }
  }
>
  <li
    key="0"
  >
    <Artist
      artist={
        Object {
          "name": "lady",
        }
      }
    />
  </li>
</ul>
`;
